lualib/rspamadm/dmarc_report.lua

```diff
@@ -345,7 +345,7 @@ end
 local function validate_reporting_domain(reporting_domain)
   -- Now check the domain policy
   -- DMARC domain is a esld for the reporting domain
-  local dmarc_domain = rspamd_util.get_tld(reporting_domain)
+  local dmarc_domain = reporting_domain
```

Comment:
The comment says that `DMARC domain is a esld for the reporting domain` but the code does the opposite thing now. And I clearly can say that such a change is incorrect by standard. We need to check the policy for eSLD and then check the subdomains policy.